docs/reference-docs/domain_models/product_types.md

```diff
@@ -0,0 +1,56 @@
+
+!!! abstract inline end "Fixed Inputs"
+    When a hard coded value is stored on product model, like `Node_Type` is here, it is called a Fixed Input. Read more about Fixed Inputs [here](model_attributes.md#fixed-inputs)
+
+Breaking this product down a bit more, we see 3 classes, `NodeInactive`, `NodeProvisioning`, and finally `Node`. These three classes are built off of each-other, with the lowest level class (`NodeInactive`) based off of the `SubscriptionModel` base class. Each class has two simple attributes, one is the Fixed Input of `Node_Type`, and the other is the root product block `node`. Each one of these classes represents the `Node` product in its various lifecycle states, which are defined here in the `SubscriptionLifecycle` enum:
+
+::: orchestrator.types.SubscriptionLifecycle
+    options:
+        heading_level: 3
+
+To fully understand the Subscription Model, it's best to look at the `SubscriptionModel` itself in the code. Here you can also see the various methods available for use on these Subscription instances when you are using them in your workflow code:
+
+::: orchestrator.domain.base.SubscriptionModel
+    options:
+        heading_level: 3
+
+It is also quite helpful to see how the Product Type is stored in the database—To see this, look at the `ProductTable` model as it shows all of the attributes stored in the database to store your WFO products:
+
+::: orchestrator.db.models.ProductTable
+    options:
+        heading_level: 3
+
+## Subscription Model Registry
+
+When you define a Product Type as a domain model in python, you also need to register it in the subscription model registry, by using the `SUBSCRIPTION_MODEL_REGISTRY` dictionary, like is shown here in the [example workflow orchestrator](https://github.com/workfloworchestrator/example-orchestrator):
+
+```python
+{% include 'https://raw.githubusercontent.com/workfloworchestrator/example-orchestrator/master/products/__init__.py' %}
+```
+
+## Creating Database Migrations
+
+After defining all of the components of a Product type, you'll also need to create a database migration to properly wire-up the product in the orchestrator's database. A migration file for this example Node model looks like this:
+
+??? example "Example: `example-orchestrator/migrations/versions/schema/2023-10-27_a84ca2e5e4db_add_node.py`"
+    ```python linenums="1"
+    {% include 'https://raw.githubusercontent.com/workfloworchestrator/example-orchestrator/master/migrations/versions/schema/2023-10-27_a84ca2e5e4db_add_node.py' %}
+    ```
+
+Thankfully, you don't have to write these database migrations by hand, you can simply use the `main.py db migrate-domain-models` command that is part of the [orchestrator CLI, documented here.](../cli.md#migrate-domain-models)
+
+## Automatically Generating Product Types
```

Comment:
might be better to move this above `Creating Database Migrations` because you want to create products before creating migrations